documentation/CAMARA-API-Design-Guide.md

```diff
@@ -0,0 +1,1165 @@
+   for example, PBKDF2, bcrypt and scrypt algorithms.
+
+4. **Information must not be exposed in the URLs**
+Usernames, passwords, session tokens, and API keys should not appear in the URL, as this can be captured in web server logs, making them easily exploitable. For example, this URL (```https://api.domain.com/user-management/users/{id}/someAction?apiKey=abcd123456789```) exposes the API key. Therefore, never use this kind of security.
+
+5. **Authentication and authorization must be considered**
+   CAMARA uses the authentication and authorization protocols and flows as described in the [Camara Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md).
+
+6. **Add request time flags should be considered**. 
+Along with other request parameters, a request timestamp can be added as a custom HTTP header in API requests.
+   The server will compare the current timestamp with the timestamp of the request
+   and will only accept the request if it is within a reasonable time frame
+   (1–2 minutes maybe).
+   - This will prevent very basic replay attacks from people trying to hack your system without changing this timestamp.
+
+7. **Entry params validation**
+ Validates the parameters of the request in the first step before it reaches the application logic.
+   Put strong validation checks and reject the request immediately if validation fails.
+   In the API response,
+   send relevant error messages and an example of the correct input format to improve the user experience.
+
+### 6.2. Security definition
+
+In general, all APIs must be secured to ensure who has access to what and for what purpose.
+CAMARA uses OIDC and CIBA for authentication and consent collection and to determine whether the user has,
+e.g. opted out of some API access.
+
+The [Camara Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#purpose) defines that a single purpose is encoded in the list of scope values. The purpose values are defined by W3C Data Privacy Vocabulary as indicated in the [Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#purpose-as-a-scope).
+
+### 6.3. Expressing Security Requirements
+
+Security requirements of an API are expressed in OpenAPI through [Security Requirement Objects](https://spec.openapis.org/oas/v3.0.3#security-requirement-object).
+
+The following is an example of how to use the `openId` security scheme defined above as described in [Use of security property](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#use-of-security-property):
+
+```yaml
+paths:
+  {path}:
+    {method}:
+      ...
+      security:
+        - openId:
+            - {scope}
+```
+

```

Comment:
hmm, this is more a repetition. CAMARA requires the usage of the openIdConnection security scheme in other places. So, what is the additional requirement (see heading)?